qtfred/src/mission/dialogs/ShipTextureReplacementDialogModel.cpp

```diff
@@ -0,0 +1,642 @@
+												do {
+													end--;
+												} while (end != ii && !stricmp(end->ship_name, Ships[_editor->cur_ship].ship_name));
+												if (end == ii)
+													break;
+												texture_set(&(*ii), &(*end));
+											}
+										}
+
+										if (end != Fred_texture_replacements.end())
+											Fred_texture_replacements.erase(end, Fred_texture_replacements.end());
+
+										// now put the new entries on the end of the list
+										texture_replace tr;
+										strcpy_s(tr.old_texture, defaultTextures[index].c_str());
+										strcpy_s(tr.new_texture, fullName.c_str());
+										strcpy_s(tr.ship_name, Ships[_editor->cur_ship].ship_name);
+										tr.new_texture_id = -1;
+										tr.from_table = false;
+
+										// assign to global FRED array
+										Fred_texture_replacements.push_back(tr);
+										_editor->missionChanged();
+									}
+									else {
+										object* objp = nullptr;
+										objp = GET_FIRST(&obj_used_list);
+										while (objp != END_OF_LIST(&obj_used_list)) {
+											if (((objp->type == OBJ_SHIP) || (objp->type == OBJ_START)) &&
+												(objp->flags[Object::Object_Flags::Marked])) {
+												Assert((objp->type == OBJ_SHIP) || (objp->type == OBJ_START));
+												auto shipp = &Ships[get_ship_from_obj(objp)];
+												end = Fred_texture_replacements.end();
+												for (ii = Fred_texture_replacements.begin(); ii != end; ++ii)
+												{
+													if (!stricmp(ii->ship_name, shipp->ship_name))
+													{
+														do {
+															end--;
+														} while (end != ii && !stricmp(end->ship_name, shipp->ship_name));
+														if (end == ii)
+															break;
+														texture_set(&(*ii), &(*end));
+													}
+												}
+
+												if (end != Fred_texture_replacements.end())
+													Fred_texture_replacements.erase(end, Fred_texture_replacements.end());
+
+												// now put the new entries on the end of the list
+												texture_replace tr;
+												strcpy_s(tr.old_texture, defaultTextures[index].c_str());
+												strcpy_s(tr.new_texture, fullName.c_str());
+												strcpy_s(tr.ship_name, shipp->ship_name);
+												tr.new_texture_id = -1;
+												tr.from_table = false;
+
+												// assign to global FRED array
+												Fred_texture_replacements.push_back(tr);
+												_editor->missionChanged();
+											}
+
+											objp = GET_NEXT(objp);
+										}
+									}
+								}
+								else {
+									auto button = _viewport->dialogProvider->showButtonDialog(DialogType::Error, "Missing Texture", "FRED was unable to find %s \n Skipping",
+										{ DialogButton::Ok });
+									if (button == DialogButton::Ok) {
+										return;
+									}
+								}
+
+							}
+						}
+						else {
+							error_display(0, "You told us to replace the -misc map with inherited data but didnt change the main texture name. Ignoring -misc map change");
```

Comment:
I would phrase this in a different way.  Suggestion - "Cannot use inherited data without changing the main texture name.  Ignoring -misc map change."